### PR TITLE
feat: supporting duplicate detection

### LINKED ISSUE #4: support Service Bus duplicate detection
https://docs.microsoft.com/en-us/azure/service-bus-messaging/duplicate-detection

### PR DESCRIPTION
**Pull request by keikumata**
Issue: #4

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -108,3 +124,23 @@ err := publisher.Publish(
...
+    Command: Command{
+        Id:            uuid.New(),
+        DestinationId: "destination",
+    },
+}
+messageID := "someMessageIDWithBusinessContext"
+err := publisher.Publish(
+    ctx,
+    cmd,
+    pubsub.SetMessageID(messageID),
```
Comment (serbrech) [inline README.md:135]: so, the messageID is set as part of the Publish, and not on the event struct itself.
I think I'm fine with that.
what do you think of saying that if the message has a property MessageID, we use that by convention

Comment (serbrech) [reply README.md:135]: keep the Publish func option param, but add a convention too

Comment (serbrech) [reply README.md:135]: separate PR, obviously :)

Comment (keikumata) [reply README.md:135]: i thought about that actually, but i thought parsing things underneath will cause more confusion and potential bugs. I think forcing the publisher to explicitly set message ID is clearer

File: README.md [hunk 2]
```diff
@@ -80,6 +80,22 @@ publisher, _ := pubsub.NewPublisher(
 )
 ```
 
+### Initializing a publisher with duplication detection
+Duplication detection cannot be enabled on topics that already exist.
```
Comment (serbrech) [inline README.md]: Should we always enable duplicate detection, by default?
what's the downside?

Comment (serbrech) [reply README.md]: ah, can you change the dedup window after topic creation?

Comment (keikumata) [reply README.md]: unfortunately you can't change the dedupe window after topic creation (SB-side limitation). we discussed offline and decided that we will allow for using duplicate detection but will park the decision of whether to enable it by default.

**Pull request merged**